tests/core/test_storage_migrations.py

```diff
@@ -8,9 +8,72 @@
 from kinto.core.cache import postgresql as postgresql_cache
 from kinto.core.permission import postgresql as postgresql_permission
 from kinto.core.storage import postgresql as postgresql_storage
+from kinto.core.storage.postgresql.migrator import Migrator
 from kinto.core.testing import skip_if_no_postgresql
 
 
+class MigratorTest(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.migrator = Migrator()
+        here = os.path.abspath(os.path.dirname(__file__))
+        migrations_directory = os.path.join(here, 'migrations')
+        self.migrator.migrations_directory = migrations_directory
+
+    def test_schema_is_created_if_no_version(self):
+        self.migrator.schema_version = 6
+        with mock.patch.object(self.migrator, 'create_schema') as create_schema:
+            self.migrator.create_or_migrate_schema()
+        self.assertTrue(create_schema.called)
+
+    def test_schema_is_not_touched_if_already_current(self):
+        self.migrator.schema_version = 6
+        # Patch to keep track of SQL files executed.
+        with mock.patch.object(self.migrator, '_execute_sql_file') as execute_sql:
+            with mock.patch.object(self.migrator, '_get_installed_version') as installed_version:
+                installed_version.return_value = 6
+                self.migrator.create_or_migrate_schema()
+                self.assertFalse(execute_sql.called)
+
+    def test_migration_file_is_executed_for_every_intermediary_version(self):
+        self.migrator.schema_version = 6
+
+        versions = [6, 5, 4, 3, 3]
+        self.migrator._get_installed_version = lambda: versions.pop()
+
+        with mock.patch.object(self.migrator, '_execute_sql_file') as execute_sql:
+            self.migrator.create_or_migrate_schema()
+        sql_called = execute_sql.call_args_list[-3][0][0]
+        self.assertIn('migrations/migration_003_004.sql', sql_called)
+        sql_called = execute_sql.call_args_list[-2][0][0]
+        self.assertIn('migrations/migration_004_005.sql', sql_called)
+        sql_called = execute_sql.call_args_list[-1][0][0]
+        self.assertIn('migrations/migration_005_006.sql', sql_called)
+
+    def test_migration_files_are_listed_if_ran_with_dry_run(self):
+        self.migrator.schema_version = 6
+
+        versions = [6, 5, 4, 3, 3]
+        self.migrator._get_installed_version = lambda: versions.pop()
+
+        with mock.patch('kinto.core.storage.postgresql.migrator.logger') as mocked:
```

Comment:
Arguably, perhaps all logging you should be captured. E.g.

```python
    @mock.patch('kinto.core.storage.postgresql.migrator.logger')
    def setUp(self, mocked_logger):
        self.migrator = Migrator()
        self.mocked_logger = mocked_logger
```

Reply:
Rethinking how logging is done in the test suite is outside the scope of this PR, although I agree it needs work.